### ISSUE TITLE
Building on MinGW32 fails

### ISSUE THREAD

**Issue by alvaro-cuesta** [DaGenix/rust-crypto#369]
```
error: failed to run custom build command for `rust-crypto v0.2.36`
Process didn't exit successfully: `C:\Users\kaoD\rustocin\target\debug\build\rust-crypto-0caf103d2b9a229d\build-script-build` (exit code: 101)
--- stdout
TARGET = Some("i686-pc-windows-gnu")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
TARGET = Some("i686-pc-windows-gnu")
debug=true opt-level=0
TARGET = Some("i686-pc-windows-gnu")
HOST = Some("i686-pc-windows-gnu")
CFLAGS_i686-pc-windows-gnu = None
CFLAGS_i686_pc_windows_gnu = None
HOST_CFLAGS = None
CFLAGS = None
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-g" "-m32" "-o" "C:\\Users\\kaoD\\rustocin\\target\\debug\\build\\rust-crypto-0caf103d2b9a229d\\out\\src\\util_helpers.o" "-c" "src/util_helpers.c"


failed to execute command: El sistema no puede encontrar el archivo especificado. (os error 2)
Is `cc` not installed? (see https://github.com/alexcrichton/gcc-rs#compile-time-requirements for help)
```

According to the [link in the error message](https://github.com/alexcrichton/gcc-rs#compile-time-requirements) `gcc-rs` should be using `gcc` under my system. [This hardcoded `"cc"` in `build.rs`](https://github.com/DaGenix/rust-crypto/blob/master/build.rs#L35) seems to be the problem.

Setting envvar `CC=gcc` fixes it, but it should probably use the default `gcc-rs` strategy for that else branch.

**Comment by PhilKershaw** [comment]
This is not so much a bug as a missing dependency. I was experiencing the same problem but a little searching around the web pointed me towards installing this:

http://win-builds.org/doku.php/download_and_installation_from_windows

Also ensure the 'bin' directory residing within the installation path is present in your PATH - as usual - and away you go. You'll find 'cc.exe' within the bin directory.

**Comment by zoumi** [comment]
I'm suffering from this error too.Thanks for the tip:"Setting envvar CC=gcc fixes it."@alvaro-cuesta.

**Comment by jcbritobr** [comment]
I think gcc should be use as default also.

**Comment by nano3k0a** [comment]
How do I set the envvar CC=gcc under Windows 7/8.1/10???

**Comment by zoumi** [comment]
http://www.computerhope.com/issues/ch000549.htm
@Trodis

**Comment by rdrsss** [comment]
Yep, just hit this problem myself on Windows 10. I'll set the env var for now, is there a proper fix coming for this?

**Comment by hk-51** [comment]
> Setting envvar 'CC=gcc' fixes it,

@alvaro-cuesta  What does it means? How can I do it too?

**Comment by kpcyrd** [comment]
I ran across this problem in appveyor, where CC doesn't seem to be available with `TARGET=x86_64-pc-windows-gnu`. I found the solution in the [appveyor.yml][0] of this project:

```yaml
environment:
  matrix:
    - TARGET: nightly-x86_64-pc-windows-msvc
    - TARGET: nightly-x86_64-pc-windows-gnu
      MSYS_BITS: 64
  [...]
install:
[...]
- if defined MSYS_BITS set PATH=C:\msys64\mingw%MSYS_BITS%\bin;C:\msys64\usr\bin;%PATH%
```

In thise file, every mingw build has `MSYS_BITS` defined which adds the mingw bin folder to the PATH prior to starting the build. This fixes the mingw build without breaking the -msvc builds.

[0]: https://github.com/DaGenix/rust-crypto/blob/cc1a5fde1ce957bd1a8a2e30169443cdb4780111/appveyor.yml

**Comment by steelx** [comment]
how do you set cc=gcc ? what does it mean.
I have installed migwin. nothing resolved after.

![image](https://user-images.githubusercontent.com/3184210/89128976-fc78e680-d516-11ea-95f0-8b641d29cf4f.png)

**Comment by newpavlov** [comment]
@steelx 
https://rustsec.org/advisories/RUSTSEC-2016-0005.html

**Comment by steelx** [comment]
> @steelx 
> https://rustsec.org/advisories/RUSTSEC-2016-0005.html

I'm using openssl. And internally it uses this it seems

**Comment by newpavlov** [comment]
No, OpenSSL does not use `rust-crypto`, neither the other way around. If you have issues with OpenSSL bindings try asking in respective repositories, e.g. in https://github.com/sfackler/rust-openssl.
